lesson3/Main.java

```diff
@@ -0,0 +1,77 @@
+package lesson3;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
+
+public class Main {
+    public static void main(String[] args) {
+        List<Student> students = generateStudents();
+        List<Student> filteredStudents = students.stream()
+                .filter(student -> student.address().city().equals("New York"))
+                .filter(student -> student.age() > 15)
+                .toList();
+
+        List<StudentGradeRecord> studentGradeRecord = filteredStudents.stream()
+                .flatMap(student ->
+                        student.listGradle().stream()
+                                .map(grade -> new StudentGradeRecord(student.name(), student.school(), grade.name(), grade.score()))
+                )
+                .sorted(Comparator.comparingDouble(StudentGradeRecord::score).reversed())
+                .limit(3)
+                .toList();
+
+        studentGradeRecord.forEach(System.out::println);
+    }
+
+    public static List<Student> generateStudents() {
+        List<Student> students = new ArrayList<>();
+
+        String[] address = new String[]{"New York", "Dallas", "Toronto", "Los Angeles", "Tampa"};
+        int age;
+
+        for (int i = 0; i < 10; i++) {
+            String studentName = "name " + i;
+            String schoolName = "school " + i;
+            List<Grade> listGrade = new ArrayList<>();
+            String city;
+            if (i % 2 == 0) {
+                city = address[0];
+                if (i != 8) {
+                    age = 18;
+                } else {
+                    age = 14;
+                }
```

Comment:
To generate dummy data you may use `new Random().nextInt()`. As a result, your code may look simpler.